File: C++/Data Structure/001. Bubble Sort.cpp
#include<iostream>
using namespace std;
void PrintArray(int Arr[], int size){
    for(int i=0;i<size ; i++){
        cout<<Arr[i]<<"\t";
    }
    cout<<endl;
}
int main(){
    int i, j, n, temp;
    cout<<"Enter Size of Array: ";
    cin>>n;
    int arr[n];
    cout<<"Enter "<<n<<" Number: "<<endl;
    for(i=0;i<n;i++){
        cin>>arr[i];
    }
    cout<<"Before Bubble Sort: "<<endl;
    PrintArray(arr,n);
    //Applying Bubble Sort
    for(i=0 ;i<n;i++){
        for(j=0 ; j<n-i; j++){
            if(arr[j]>arr[j+1]){
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
    cout<<"After Bubble SOrt"<<endl;
    PrintArray(arr,n);
}

File: Java/Data Structure/001. Bubble Sort.java
import java.util.*;
public class BubbleSort{
    static int i, j, n, temp;
    static void PrintArray(int Arr[], int size){
        for(i=0; i<size; i++){
            System.out.print(Arr[i]+"\t");
        }
        System.out.println();
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter Size of Array:");
        n = sc.nextInt();
        int arr[] = new int[n];
        System.out.println("Enter "+n+" Number:");
        for(i=0;i<n;i++){
            arr[i] = sc.nextInt();
        }
        System.out.println("Before Bubble Sort: ");
        PrintArray(arr, n);
        //Applying Bubble Sort
        for(i=0 ;i<n; i++){
            for(j=0; j<n-i-1; j++){
                if(arr[j]>arr[j+1]){
                    temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
        System.out.println("After Bubble Sort: ");
        PrintArray(arr, n);
    }
}
